### PR TITLE
Apply XML parser security features

### PR DESCRIPTION
**Pull request by pjfanning**
Copied over some POI code to protect the DOM parser from XML entity expansion attacks.
The POI DocumentHelper code creates Namespace Aware parsers and these don't work in current excel-streaming-reader code (the XPath expressions don't have namespace prefixes, etc.).

### CODE REVIEW

**Comment by pjfanning** [comment]
@monitorjbl could you review this, if you get a chance?

**Comment by monitorjbl** [comment]
There are two outstanding reviews already on this PR

**Comment by pjfanning** [comment]
I don't see the review items for this PR, would you be able to add the review comments again?

**Comment by pjfanning** [comment]
@monitorjbl can you reconsider merging this?

**Comment by monitorjbl** [comment]
@pjfanning I ended up implementing this over the weekend, but I did it differently from your PR. I didn't realize what an entity expansion attack could do or how serious it was when you opened this or I would have jumped on this a lot sooner.

File: src/main/java/com/monitorjbl/xlsx/XmlUtils.java [hunk 1]
```diff
@@ -31,4 +41,31 @@ public static NodeList searchForNodeList(Document document, String xpath) {
...
+    String[] classNames = new String[]{"com.sun.org.apache.xerces.internal.util.SecurityManager", "org.apache.xerces.util.SecurityManager"};
+
+    for (String securityManagerClassName : classNames) {
+      try {
+        Object mgr = Class.forName(securityManagerClassName).newInstance();
+        Method setLimit = mgr.getClass().getMethod("setEntityExpansionLimit", Integer.TYPE);
+        setLimit.invoke(mgr, Integer.valueOf(4096));
+        dbf.setAttribute("http://apache.org/xml/properties/security-manager", mgr);
+        return;
+      } catch (Throwable t) {
```
[the remark this answers is not in the document]
Comment (pjfanning) [reply src/main/java/com/monitorjbl/xlsx/XmlUtils.java]: this is a copy/paste from the POI codebase - it updates the a xerces specific setting while not preventing users from choosing another parser

**Pull request closed without merging**